packages/firestore/test/unit/specs/spec_test_components.ts

```diff
@@ -66,12 +67,17 @@ export class MockMemoryPersistence extends MemoryPersistence {
     ) => PersistencePromise<T>
   ): Promise<T> {
     if (this.injectFailures) {
-      return Promise.reject(
-        new IndexedDbTransactionError(new Error('Simulated retryable error'))
-      );
-    } else {
-      return super.runTransaction(action, mode, transactionOperation);
+      if (this.injectFailures[action as PersistenceAction] === undefined) {
```

Comment:
Consider saving the result of `this.injectFailures[action as PersistenceAction]` to a local variable instead of accessing it twice via this long dictionary access syntax.

Reply:
Done.